### ISSUE TITLE
Re-aquire permission to send email after renewed contact

### ISSUE THREAD

**Issue by magnolia61** [veda-consulting/uk.co.vedaconsulting.gdpr#60]
What would be the best way to re-aquire permission to send email after someone opted out earlier?
Scenario:
1. someone is optout from mailing list or bounced mails caused email to be 'on hold'
2. after a few years same person registers for an event thus we need to get in touch again

How can we include a customfield in the registration or a civirule to:
1. remove the 'on hold' flag from the email
2. remove possible 'Do not email' flag for this person

Could this extension play a role in this?

**Comment by veda-consulting** [comment]
@magnolia61 what we're advising is that the communication preferences page that this extension provides be added to all thank you or end of journey processes. So for example in your case someone signs up for an event, they pay and they're taken through to the thank you page. This thank you page should include something like "Thanks, by the way we'd love to keep in touch, please take a minute to update your communication preferences" with a button through to a checksumed link to the comms preferences page. This will give them the oppurtunity to opt back in etc. And this is probably the most GDPR compliant way to do it rather then carrying out implicit re-opt in etc.

**Comment by magnolia61** [comment]
Thanks for your quick response, I understand most of what you say. But when they register for an event or membership or make a contribution, we cannot even send them their receipt of invoice as the email-address is still 'on hold'. (I agree that for bulk mailings it should be re-asked). What is your advise on that scenario?

**Comment by veda-consulting** [comment]
@magnolia61 thats an interesting point, my understanding was CiviCRM would remove this on hold but I might be wrong - its not a scenario I've trialed. If it doesn't get removed. I think the correct action is that it does (worth testing what happens if the email address is updated during registration via a checksum link). We could put that into this extension i.e. remove on hold if there are certain actions against the same email address i.e. registration for event, donation etc. 

Also worth noting that we have factored in amending scheduled reminders system at some point to take into account bulk email preferences, as a lot of organisations are using scheduled reminders for automated journeys but at present shceduled reminders do not have any indicator to determine if they are transactional emails or not.
